Removing unused yaml file & putting tmhmm.py back in default env

komorn opened the pull request:
- Removed unused yaml file: Deleted the `tmhmm.yaml` file as it was not used anywhere in the pipeline, which I didn't realize until now 
- Restored tmhmm dependency in the default environment: reintroduced the tmhmm.py dependency back into the `default.yaml` environment. This dependency was removed inadequately in a previous change.
- Python version condition: Using condition <3.12 on python version because of torch version conditions for SignalP

komorn committed:
Removing unused yaml file & putting tmhmm.py back in default env

tomas-pluskal commented:
I don't like having python<3.12 condition in the default environment - it will create problems for future upgrades of the dependencies. In the other pull request you moved signalp into its own environment, so why don't you follow on that one?

komorn commented:
@tomas-pluskal I added tmhmm.py to default and removed `tmhmm.yaml` file in the other pull request just now. I didn't follow up there because there were no comments and it hadn't been merged so I proceeded with these updates in a different pull request.
